### PR TITLE
[Merged by Bors] - feat: port `Algebra.Ring.Equiv`

### PR DESCRIPTION
**Pull request by ADedecker**
See discussion [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/norm_cast.3A.20badly.20shaped.20lemma) and [there](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Algebra.2ERing.2EEquiv.20help.20request)

I used #835 as a reference

- [x] depends on: #1150

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1150
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1150~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Ring/Equiv.lean [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Group.Opposite
+import Mathlib.Algebra.Hom.Ring
+import Mathlib.Logic.Equiv.Set
+
+/-!
+# (Semi)ring equivs
+
+In this file we define extension of `equiv` called `ring_equiv`, which is a datatype representing an
```
Comment (jcommelin) [inline Mathlib/Algebra/Ring/Equiv.lean:19]:
```suggestion
In this file we define extension of `equiv` called `RingEquiv`, which is a datatype representing an
```
Please check all other comments and docstrings for similar changes.

Comment (ADedecker) [reply Mathlib/Algebra/Ring/Equiv.lean]: Sorry, that should be fixed now

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**